tsdb/querier.go

```diff
@@ -891,7 +891,7 @@ type mergedStringIter struct {
 }
 
 func (m *mergedStringIter) Next() bool {
-	if (!m.aok && !m.bok) || (m.Err() != nil) {
+	if !m.aok && !m.bok {
```

Comment:
We should not ignore the error here. This change should be reverted. That said, https://github.com/prometheus/prometheus/pull/12132 made some changes here recently. Can you update your branch with latest changes in the main instead?